crates/uv/src/commands/pip_list.rs

```diff
@@ -70,6 +79,205 @@ pub(crate) fn pip_list(
+            let client = FlatIndexClient::new(&client, cache);
+            let entries = client.fetch(index_locations.flat_index()).await?;
+            FlatIndex::from_entries(entries, tags)
+        };
+
+        let build_isolation = BuildIsolation::Isolated;
+
+        // Create a shared in-memory index.
+        let index = InMemoryIndex::default();
+
+        // Track in-flight downloads, builds, etc., across resolutions.
+        let in_flight = InFlight::default();
+
+        let resolve_dispatch = BuildDispatch::new(
+            &client,
+            cache,
+            &interpreter,
+            &index_locations,
+            &flat_index,
+            &index,
+            &in_flight,
+            setup_py,
+            &config_settings,
+            build_isolation,
+            &no_build,
+            &no_binary,
+        )
+        .with_options(OptionsBuilder::new().exclude_newer(exclude_newer).build());
+
+        // Build all editable distributions. The editables are shared between resolution and
+        // installation, and should live for the duration of the command. If an editable is already
+        // installed in the environment, we'll still re-build it here.
+        let editable_wheel_dir;
+        let editables = if editables.is_empty() {
+            vec![]
+        } else {
+            editable_wheel_dir = tempdir_in(venv.root())?;
+            build_editables(
+                &editables,
+                editable_wheel_dir.path(),
+                cache,
+                &interpreter,
+                tags,
+                &client,
+                &resolve_dispatch,
+                printer,
+            )
+            .await?
+        };
+
+        let options = OptionsBuilder::new()
+            .resolution_mode(resolution_mode)
+            .prerelease_mode(prerelease_mode)
+            .dependency_mode(dependency_mode)
+            .exclude_newer(exclude_newer)
+            .build();
+
+        // Resolve the requirements.
+        let resolution = match resolve(
+            requirements,
+            constraints,
+            overrides,
+            project,
+            &editables,
+            &site_packages,
+            &reinstall,
+            &upgrade,
+            &interpreter,
+            tags,
+            markers,
+            &client,
+            &flat_index,
+            &index,
+            &resolve_dispatch,
+            options,
+            printer,
+        )
+        .await
+        {
+            Ok(resolution) => Resolution::from(resolution),
+            Err(Error::Resolve(uv_resolver::ResolveError::NoSolution(err))) => {
+                let report = miette::Report::msg(format!("{err}"))
+                    .context("No solution found when resolving dependencies:");
+                eprint!("{report:?}");
+                return Ok(ExitStatus::Failure);
+            }
+            Err(err) => return Err(err.into()),
+        };
+
+        results = results
```

Comment:
Finally, we filter `results` based on whether there is a more recent version of the package available.